File: fastapi/purchase-microservice/model/purchase_event.py
from bson import ObjectId
from pydantic import BaseModel, Field

from .py_object_id import PyObjectId


class PurchaseEventModel(BaseModel):
    id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")
    index: str = Field(...)
    name: str = Field(...)
    gender: str = Field(...)
    email: str = Field(...)
    phone: str = Field(...)
    dateOfBirth: str = Field(...)
    houseNumber: str = Field(...)
    price: str = Field(...)
    sqFt: str = Field(...)
    bedrooms: str = Field(...)
    bathrooms: str = Field(...)
    region: str = Field(...)

    class Config:
        allow_population_by_field_name = True
        arbitrary_types_allowed = True
        json_encoders = {ObjectId: str}
        schema_extra = {
            "example": {
                "index": "1"
            }
        }

File: nodejs/purchase-microservice/model/purchase_event.js
const mongoose = require('mongoose');

const purchaseEventSchema = new mongoose.Schema({
    index: {
        required: true,
        type: String
    },
    name: {
        required: true,
        type: String
    },
    gender: {
        required: true,
        type: String
    },
    email: {
        required: true,
        type: String
    },
    phone: {
        required: true,
        type: String
    },
    dateOfBirth: {
        required: true,
        type: String
    },
    houseNumber: {
        required: true,
        type: String
    },
    price: {
        required: true,
        type: String
    },
    sqFt: {
        required: true,
        type: String
    },
    bedrooms: {
        required: true,
        type: String
    },
    bathrooms: {
        required: true,
        type: String
    },
    region: {
        required: true,
        type: String
    }
})

module.exports = mongoose.model('PurchaseEvent', purchaseEventSchema)
